src/Graphics/src/Graphics/Platforms/Windows/PlatformImage.cs

```diff
@@ -153,10 +154,38 @@ public IImage ToImage(int width, int height, float scale = 1f)
 		public static IImage FromStream(Stream stream, ImageFormat format = ImageFormat.Png)
 		{
 			var creator = PlatformGraphicsService.Creator;
+			
 			if (creator == null)
+			{
 				throw new Exception("No resource creator has been registered globally or for this thread.");
+			}
+
+			CanvasBitmap bitmap;
+
+			if (stream.CanSeek)
+			{
+				global::Windows.Foundation.IAsyncOperation<CanvasBitmap> bitmapAsync = CanvasBitmap.LoadAsync(creator, stream.AsRandomAccessStream());
+				bitmap = bitmapAsync.AsTask().GetAwaiter().GetResult();
+			}
+			else
+			{
+				byte[] bytes = ArrayPool<byte>.Shared.Rent(100 * 1024); // 100 kB.
```

Comment:
Can `stream.Length` be used here? Or will that throw an exception in some cases?

Reply:
Yeah, just a random web search, certain streams will throw `NotSupportedException`:
* https://stackoverflow.com/q/3373579